File: auth-application/src/main/java/com/example/auth/application/port/in/ListMySessionsUseCase.java
package com.example.auth.application.port.in;

import com.example.auth.domain.common.TenantId;
import com.example.auth.domain.common.UserId;
import java.time.Instant;
import java.util.List;
import java.util.UUID;

public interface ListMySessionsUseCase {

    List<SessionView> list(TenantId tenantId, UserId userId);

    record SessionView(
            UUID sessionId,
            String deviceLabel,
            String ipAddress,
            Instant issuedAt,
            Instant lastUsedAt,
            Instant expiresAt) {
    }
}

File: auth-application/src/main/kotlin/com/example/auth/application/port/in/ListMySessionsUseCase.kt
package com.example.auth.application.port.`in`

import com.example.auth.domain.common.TenantId
import com.example.auth.domain.common.UserId
import java.time.Instant
import java.util.UUID

interface ListMySessionsUseCase {

    fun list(tenantId: TenantId, userId: UserId): List<SessionView>

    @JvmRecord
    data class SessionView(
        val sessionId: UUID,
        val deviceLabel: String?,
        val ipAddress: String?,
        val issuedAt: Instant,
        val lastUsedAt: Instant?,
        val expiresAt: Instant,
    )
}
